monai/losses/aucm_loss.py

```diff
@@ -0,0 +1,172 @@
+            auto: if True, automatically calculate imbalance ratio. Defaults to True.
+
+        Raises:
+            ValueError: When y_pred and num_classes have different sizes.
+            ValueError: When y_true and num_classes have different sizes.
+            ValueError: When y_pred and y_true have different sizes.
+        """
+        if y_pred.shape[1] != self.num_classes:
+            raise ValueError(
+                f"y_pred.shape[1] must be {self.num_classes}, equal to num_classes ,but is {y_pred.shape[1]}"
+            )
+
+        if y_true.shape[1] != self.num_classes:
+            raise ValueError(
+                f"y_true.shape[1] must be {self.num_classes}, equal to num_classes ,but is {y_true.shape[1]}"
+            )
+
+        if self.sigmoid:
+            y_pred = torch.sigmoid(y_pred)
+
+        n_pred_ch = y_pred.shape[1]
+        if self.softmax:
+            if n_pred_ch == 1:
+                warnings.warn("single channel prediction, `softmax=True` ignored.")
+            else:
+                y_pred = torch.softmax(y_pred, dim=1)
+
+        if self.other_act is not None:
+            y_pred = self.other_act(y_pred)
+
+        if self.to_onehot_y:
+            if n_pred_ch == 1:
+                warnings.warn("single channel prediction, `to_onehot_y=True` ignored.")
+            else:
+                y_true = one_hot(y_true, num_classes=n_pred_ch)
+
+        if y_pred.shape != y_true.shape:
+            raise ValueError(
+                f"y_pred and y_true must have the same shape. y_pred.shape: {y_pred.shape}, y_true.shape: {y_true.shape}"
+            )
+
+        total_loss = 0
+
+        # let the device of the weights same as input
+        self.a.to(y_pred.device)
+        self.b.to(y_pred.device)
+        self.alpha.to(y_pred.device)
+
+        for idx in range(self.num_classes):
+            if len(y_pred[:, idx].shape) == 1:
+                y_pred = y_pred[:, idx].reshape(-1, 1)
+            if len(y_true[:, idx].shape) == 1:
+                y_true = y_true[:, idx].reshape(-1, 1)
+            if auto or not self.p:
+                self.p[idx] = (y_true == 1).sum() / y_true.shape[0]
+
+            loss = (
+                (1 - self.p[idx]) * torch.mean((y_pred - self.a[idx]) ** 2 * (1 == y_true).type(torch.FloatTensor))
+                + self.p[idx] * torch.mean((y_pred - self.b[idx]) ** 2 * (0 == y_true).type(torch.FloatTensor))
+                + 2
+                * self.alpha[idx]
+                * (
+                    self.p[idx] * (1 - self.p[idx])
+                    + torch.mean(
+                        self.p[idx] * y_pred * (0 == y_true).type(torch.FloatTensor)
+                        - (1 - self.p[idx]) * y_pred * (1 == y_true).type(torch.FloatTensor)
+                    )
+                )
+                - self.p[idx] * (1 - self.p[idx]) * self.alpha[idx] ** 2
+            )
+
+            total_loss += loss
```

Comment:
The type of `total_loss` is not Tensor, check the output type due to https://github.com/Project-MONAI/MONAI/runs/8108464942?check_suite_focus=true, thanks!